graylog2-server/src/main/java/org/graylog/security/shares/GranteeSharesService.java

```diff
@@ -166,14 +162,8 @@ private Set<Grantee> getOwners(Map.Entry<GRN, Set<GRN>> entry) {
                 .collect(Collectors.toSet());
     }
 
-    @AutoValue
-    public static abstract class SharesResponse {
-        public abstract PaginatedList<EntityDescriptor> paginatedEntities();
-
-        public abstract Map<GRN, Capability> capabilities();
-
-        public static SharesResponse create(PaginatedList<EntityDescriptor> paginatedEntities, Map<GRN, Capability> capabilities) {
-            return new AutoValue_GranteeSharesService_SharesResponse(paginatedEntities, capabilities);
-        }
+    public record SharesResponse(
+            @JsonProperty("entities") PaginatedList<EntityDescriptor> paginatedEntities,
+            @JsonProperty("grantee_capabilities") Map<GRN, Capability> capabilities) {
```

Comment:
I think we don't need the `@JsonProperty` annotations here because the `SharesResponse` is not serialized/deserialized anywhere.